### PR TITLE
Detect yarn in Solus os

### PR DESCRIPTION
**Pull request by omarcresp**
https://github.com/vuejs/vue-cli/issues/3941

### CODE REVIEW

File: packages/@vue/cli-shared-utils/lib/env.js [hunk 1]
```diff
@@ -27,8 +27,15 @@ exports.hasYarn = () => {
     execSync('yarnpkg --version', { stdio: 'ignore' })
     return (_hasYarn = true)
   } catch (e) {
-    return (_hasYarn = false)
+    _hasYarn = false
```
Comment (pksunkara) [inline packages/@vue/cli-shared-utils/lib/env.js]:
```suggestion
```

File: packages/@vue/cli-shared-utils/lib/env.js [hunk 2]
```diff
@@ -27,8 +27,15 @@ exports.hasYarn = () => {
...
   }
+  try {
+    execSync('yarn --version', { stdio: 'ignore' })
+    return (_hasYarn = true)
+  } catch (e) {
+    _hasYarn = false
+  }
+  return _hasYarn
```
Comment (pksunkara) [inline packages/@vue/cli-shared-utils/lib/env.js]:
```suggestion
```

File: packages/@vue/cli-shared-utils/lib/env.js [hunk 3]
```diff
@@ -27,8 +27,15 @@ exports.hasYarn = () => {
...
     return (_hasYarn = true)
   } catch (e) {
-    return (_hasYarn = false)
+    _hasYarn = false
   }
+  try {
+    execSync('yarn --version', { stdio: 'ignore' })
+    return (_hasYarn = true)
+  } catch (e) {
+    _hasYarn = false
```
Comment (pksunkara) [inline packages/@vue/cli-shared-utils/lib/env.js]:
```suggestion
    return (_hasYarn = false)
```

**Comment by pksunkara** [comment]
The main reason for using `yarnpkg` is that most of OS's give preference to Hadoop YARN when you call `yarn`

**Comment by sodatea** [comment]
Hadoop YARN will throw for `yarn --version` (it only supports `yarn --version`) so won't pass our check.

Since we run `yarn` rather than `yarnpkg` everywhere else, I think it makes sense to only check `yarn`.

**Pull request closed without merging**